### ISSUE TITLE
GrabDragInteraction doesn't support touchSnag

### ISSUE THREAD

**Issue by KatieWoe** [phetsims/scenery-phet#708]
**Test device**
Dell and iPad
**Operating System**
Win 11 and iPadOS 14
**Browser**
Chrome and Safari
**Problem description**
For https://github.com/phetsims/qa/issues/721. When picking up the balloon with touch when interactive highlights are on, one of two things can happen. If picked up directly the highlight is a dotted line indicating it is grabbed. If you swipe to pick up, the out line is solid, indicating it is not grabbed, though you can manipulate the balloon as normal.

**Visuals**
![whathighlight](https://user-images.githubusercontent.com/41024075/138768944-be11cd43-fce7-4768-8b68-3034c1bf2250.gif)

Troubleshooting information:

<details>

!!!!! DO NOT EDIT !!!!!
Name: ‪Balloons and Static Electricity‬
URL: https://phet-dev.colorado.edu/html/balloons-and-static-electricity/1.5.0-rc.2/phet/balloons-and-static-electricity_all_phet.html
Version: 1.5.0-rc.2 2021-10-11 21:55:48 UTC
Features missing: applicationcache, applicationcache, touch
Flags: pixelRatioScaling
User Agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/95.0.4638.54 Safari/537.36
Language: en-US
Window: 1280x649
Pixel Ratio: 1.5/1
WebGL: WebGL 1.0 (OpenGL ES 2.0 Chromium)
GLSL: WebGL GLSL ES 1.0 (OpenGL ES GLSL ES 1.0 Chromium)
Vendor: WebKit (WebKit WebGL)
Vertex: attribs: 16 varying: 30 uniform: 4096
Texture: size: 16384 imageUnits: 16 (vertex: 16, combined: 32)
Max viewport: 32767x32767
OES_texture_float: true
Dependencies JSON: {}

</details>

**Comment by pixelzoom** [comment]
Reviewing scenery-phet issues for Q4 planning... 

No progress in almost a year. Assigning to @kathy-phet to prioritize.

**Comment by zepumph** [comment]
Subset of https://github.com/phetsims/scenery/issues/1298

**Comment by kathy-phet** [comment]
Might have been fixed by this issue: https://github.com/phetsims/scenery/issues/1495

**Comment by jessegreenberg** [comment]
This is not fixed from https://github.com/phetsims/scenery/issues/1495, still an issue.

**Comment by samreid** [comment]
@jessegreenberg said this could be good to fix before publishing buoyancy 1.0.

**Comment by zepumph** [comment]
I'd like to also take a look at this for Buoyancy publication.

**Comment by jessegreenberg** [comment]
Replacing the PressListener with a DragListener to get touchSnag support seems to be working well. I tested by using device emulation with the dev tools which mimics touch events and touchSnag.

I committed the change to branch [grabDragInteractionTouchSnagFix](https://github.com/phetsims/scenery-phet/compare/grabDragInteractionTouchSnagFix).

@zepumph can you please review this, are you OK with it? This changed a tandem name, so if it seems OK I will continue by updating phet-io APIs and looking into migration.

**Comment by zepumph** [comment]
This is looking really great. I tested with BASE and things are excellent. I merged and updated APIs to make things easy on you. No migration needed for Friction, which was the only stable API sim that used this (buoyancy doesn't instrument this interaction.

Unfortunately (and not unsurprisingly), this doesn't add touchSnag to density/buoyancy suite. I believe it was something to the bounds of the focusable node, as reported in https://github.com/phetsims/density-buoyancy-common/issues/210#issuecomment-2287669268. From this I feel totally fine closing this issue before things are working in that sim, and to double check that this feature works for that sim in a separate issue. Thanks so much for this simple and elegant solution. Nice work!
